rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +83,293 @@ typedef struct
+    virtual void set_vertices_counts(std::vector<std::vector<int>> vertices_count) { };
+    virtual JointsData& get_joints_data(){  };
+    ImgSize& get_img_size() {return _info._img_size; }
+    std::string& get_image_name() { return _info._img_name; }
+    uint& get_image_id() { return _info._img_id; }
+    void set_img_size(ImgSize img_size) { _info._img_size = std::move(img_size); }
+    void set_img_id(uint img_id) { _info._img_id = img_id; }
+    void set_img_name(std::string img_name) { _info._img_name = std::move(img_name); }
+    void set_metadata_info(MetaDataInfo info) { _info = std::move(info); }
+    protected:
+    MetaDataInfo _info;
 };
 
 struct Label : public MetaData
 {
-    Label(int label) { _label_id = label; }
-    Label(){ _label_id = -1; }
+    Label(int label)
+    {
+        _label_ids.resize(1);
+        _label_ids[0] = label;
+    }
+    Label()
+    {
+        _label_ids.resize(1);
+        _label_ids[0] = -1;
+    }
+    std::vector<int>& get_labels() { return _label_ids; }
+    void set_labels(Labels label_ids)
+    {
+        _label_ids = std::move(label_ids);
+    }
+    protected:
+    Labels _label_ids = {}; // For label use only
 };
 
-struct BoundingBox : public MetaData
+struct BoundingBox : public Label
 {
     BoundingBox()= default;
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids)
-    {
-        _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
-    }
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids, ImgSize img_size)
+    BoundingBox(BoundingBoxCords bb_cords, Labels bb_label_ids, ImgSize img_size = ImgSize{0,0}, uint img_id = 0)
     {
         _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
-        _img_size = std::move(img_size);
+        _label_ids = std::move(bb_label_ids);
+        _info._img_size = std::move(img_size);
+        _info._img_id = img_id;
     }
-    void set_bb_cords(BoundingBoxCords bb_cords) { _bb_cords =std::move(bb_cords); }
-    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, BoundingBoxLabels bb_label_ids)
+    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, Labels bb_label_ids, ImgSize img_size = ImgSize{0,0}, uint img_id = 0)
     {
         _bb_cords_xcycwh =std::move(bb_cords_xcycwh);
-        _bb_label_ids = std::move(bb_label_ids);
+        _label_ids = std::move(bb_label_ids);
+        _info._img_size = std::move(img_size);
+        _info._img_id = img_id;
     }
-    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, BoundingBoxLabels bb_label_ids, ImgSize img_size)
+    BoundingBoxCords& get_bb_cords() { return _bb_cords; }
+    BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { return _bb_cords_xcycwh; }
+    void set_bb_cords_xcycwh(BoundingBoxCords_xcycwh bb_cords_xcycwh) { _bb_cords_xcycwh =std::move(bb_cords_xcycwh); }
+    void set_bb_cords(BoundingBoxCords bb_cords) { _bb_cords =std::move(bb_cords); }
```

Comment:
Add space after = operator in above 2 lines

Reply:
Done